client/state/sites/hooks/use-sites-sorting.ts

```diff
@@ -0,0 +1,55 @@
+import {
+	SitesTableSortOptions,
+	sitesTableSortKeys,
+	SitesTableSortKey,
+	sitesTableSortOrders,
+	SitesTableSortOrder,
+} from '@automattic/components';
+import { useAsyncPreference } from 'calypso/state/preferences/use-async-preference';
+
+const SEPARATOR = '-' as const;
+
+type SitesSorting = `${ SitesTableSortKey }${ typeof SEPARATOR }${ SitesTableSortOrder }`;
+
+const DEFAULT_SITES_SORTING = {
+	sortKey: 'updatedAt',
+	sortOrder: 'desc',
+} as const;
+
+export const parseSitesSorting = ( sorting: string ) => {
```

Comment:
@zaguiini I'd be curious if there's a better way to handle validation on an arbitrary string with some TypeScript fanciness...

Reply:
Doesn't `${ string }${ SEPARATOR }${ string } | 'none'` work?

Reply:
Why is it an arbitrary string, though?

Reply:
Refactored with e0af36fca0